### ISSUE TITLE
Lint/DuplicateMethods false positive when using `alias_method` to change/change back a method

### ISSUE THREAD

**Issue by tjschuck** [rubocop/rubocop#11273]
This is something we sometimes do as a test helper method to ensure some sort of behavior out of a method in the class to be tested.  Imagine, for example, we wanted `save` in the tested class to always raise an error when called, but we also don't want it to _stay_ modified:

```ruby
module FooTest
  def make_save_always_fail
    Foo.class_eval do
      def failed_save
        raise
      end
      alias_method :original_save, :save
      alias_method :save, :failed_save
    end

    yield
  ensure
    Foo.class_eval do
      alias_method :save, :original_save
    end
  end
end
```

If you call `rubocop --only Lint/DuplicateMethods` on that snippet, you get a false positive for the "second" definition of `save` inside the `ensure` block.

https://wasabigeek.com/blog/practical-metaprogramming-in-ruby-minitest-mock/ outlines a similar form of this.

--------

## Expected behavior

Should allow multiple `alias_method` definitions of the same name if they happen in different scopes.

## Actual behavior

See above.

## Steps to reproduce the problem

See example snippet and command above.

## RuboCop version

```
1.40.0 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.0.4) [x86_64-darwin21]
```
